encord/orm/dataset.py

```diff
@@ -117,28 +276,212 @@ def created_at(self) -> datetime:
+        each image in this group can have a different dimension. Inspect the
+        :meth:`images <encord.data.DataRow.images>` to get the height of individual images.
+        """
+        return self["height"]
+
+    @property
+    def last_edited_at(self) -> datetime:
+        return parser.parse(self["last_edited_at"])
+
+    @property
+    def file_link(self) -> Optional[str]:
+        """
+        A permanent file link of the given data asset. When stored in
+        :meth:`StorageLocation.CORD_STORAGE <encord.orm.dataset.StorageLocation.CORD_STORAGE>` this will be the
+        internal file path. In private bucket storage location this will be the full path to the file.
+        If the data type is `DataType.DICOM` then this returns None as no single file is associated with the
+        series.
+        """
+        return self["file_link"]
+
+    @property
+    def signed_url(self) -> Optional[str]:
+        """
+        The cached signed url of the given data asset. To cache the signed url, use the
+        :meth:`~encord.orm.dataset.DataRow.fetch()` function.
+        """
+        return self["signed_url"]
+
+    @property
+    def file_size(self) -> int:
+        """
+        The file size of the given data asset in bytes.
+        """
+        return self["file_size"]
+
+    @property
+    def file_type(self) -> str:
+        """
+        A MIME file type of the given data asset as a string
+        """
+        return self["file_type"]
+
+    @property
+    def storage_location(self) -> StorageLocation:
+        return self["storage_location"]
+
+    @property
+    def images(self) -> Optional[List[ImageData]]:
+        """
+        A list of the cached :class:`~encord.orm.dataset.ImageData` objects for the given data asset.
+        Fetch the images with appropriate settings in the :meth:`~encord.orm.dataset.DataRow.fetch()` function.
+        If the data type is not :meth:`DataType.IMG_GROUP <encord.constants.enums.DataType.IMG_GROUP>`
+        then this returns None.
+        """
+        return self["images"]
+
+    @property
+    def dicom_signed_urls(self) -> Optional[List[str]]:
+        """
+        The signed urls of all the DICOM files in the series of the :class:`~encord.orm.dataset.DataRow`.
+        Returns `None` if the data type is not :meth:`DataType.DICOM <encord.constants.enums.DataType.DICOM>`.
+        """
+        return self["dicom_signed_urls"]
+
+    @property
+    def is_optimised_image_group(self) -> Optional[bool]:
+        """
+        If the data type is an :meth:`DataType.IMG_GROUP <encord.constants.enums.DataType.IMG_GROUP>`,
+        returns whether this is a performance optimised image group. Returns `None` for other data types.
+        """
+        return self["is_optimised_image_group"]
+
+    def fetch(
+        self,
+        *,
+        signed_url: bool = False,
+        images: Optional[FetchImagesDataConfig] = None,
```

Comment:
Thing is, the `FetchImagesDataConfig` are the settings for the list of `ImageData` that we will fetch. `fetch_images_config` implies to me that we're fetching a config. Maybe I should call the class `FetchImagesDataSettings`?

Reply:
I see what you mean, but I think that an argument named config on a fetch method kinda implies it is the config for the fetch, not the entity being fetched. I suspect that `config` and `setting` are interchangeable in this case and depending on the view imply the same thing. (Also, if we are changing just the name so maybe `options` since is similar to the web fetch API)

I'll leave it up to you, all options are fine IMO, just `images` argument for a `fetch` method is a bit confusing.

Reply:
perhaps images_fetch_config or images_fetch_options (if there's no concern about consistent naming)